tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/AbstractHelmChartIT.java

```diff
@@ -188,4 +194,40 @@ protected boolean uninstallPlatformChart() {
             throw new AssertionError(String.format("Could not read resource file '%s'", filename), e);
         }
     }
+
+    // free disk space in K3s
+    private void cleanupK3s() throws Exception {
+        final var pruneContainerResult = helmChartContainer.execInContainer("sh", "-c", """
+                for ID in $(ctr container ls --quiet); do \
+                TASK=$(ctr task ls | grep $ID); \
+                if [ -z "$TASK" ]; then \
+                continue; \
+                fi; \
+                STATUS=$(echo $TASK | awk '{print $3}' | tr -d '[:space:]'); \
+                if [ "$STATUS" != "RUNNING" ]; then \
+                echo "Removing container $ID (status: $STATUS)"; \
+                ctr container rm $ID; \
+                fi; \
+                done""");
+        LOG.info("Removed unused containers:\n{}", pruneContainerResult.getStdout());
+        if (pruneContainerResult.getExitCode() != 0) {
+            LOG.warn("Error during removal of containers:\n{}", pruneContainerResult.getStderr());
+        }
+        // the removal will fail if the snapshot has children, so we just print a warning
+        final var pruneSnapshots = helmChartContainer.execInContainer("sh", "-c", """
+                for ID in $(ctr snapshot ls | awk '{print $1}'); do \
+                if [ "$ID" == "KEY" ]; then \
+                continue; \
+                fi; \
+                SNAPSHOT=$(ctr snapshot ls | grep $ID);\
+                if echo "$SNAPSHOT" | grep -q "Committed"; then \
+                echo "Removing snapshot $ID"; \
+                ctr snapshot rm $ID; \
+                fi; \
+                done""");
+        LOG.info("Removed unused snapshots:\n{}", pruneSnapshots.getStdout());
+        if (pruneSnapshots.getExitCode() != 0) {
+            LOG.warn("Error during removal of snapshots:\n{}", pruneSnapshots.getStderr());
+        }
+    }
```

Comment:
This code is already merged into both the `hivemq-platform-operator` and the `helm-charts` epic branches.
See https://github.com/hivemq/helm-charts/commit/490950089e39d7205fd0ea7a396173b24887eb12

Reply:
😄 Does not make it prettier

Reply:
Definitely, extremely ugly! 🙈